graphql_compiler/compiler/compiler_frontend.py

```diff
@@ -964,7 +969,9 @@ def _compile_root_ast_to_ir(schema, ast, type_equivalence_hints=None):
     # Based on the outputs context data, add an output step and construct the output metadata.
     basic_blocks.append(_compile_output_step(query_metadata_table))
     output_metadata = {
-        name: OutputMetadata(type=info.type, optional=info.optional)
+        name: OutputMetadata(type=info.type, optional=info.optional, folded=True)
+        if isinstance(info.location, FoldScopeLocation)
+        else OutputMetadata(type=info.type, optional=info.optional, folded=False)
```

Comment:
The indentation came out a bit weird here (the `if` is the same clause as the `OutputMetadata` but not indented under it). I'd either use parens to force a new line after the `:`, or do something like:
```python
name: OutputMetadata(type=info.type, optional=info.optional, folded=isinstance(info.location, FoldScopeLocation))
```